### ISSUE TITLE
Recreated scylla node fails to be attached to a scylla cluster after draining and then uncordoning a scylla k8s node

### ISSUE THREAD

**Issue by vponomaryov** [scylladb/scylla-operator#473]
**Describe the bug**
"scylla" container in a "scylla" pod which comes as replacement for the drained predecessor fails with following error:
```
Startup failed: std::runtime_error (A node with address 10.3.242.255 already exists, cancelling join. Use replace_address if you want to replace this node.)
```

List of scylla pods looks like this:
```
$ kubectl get pods -n scylla -o wide
NAME                                READY   STATUS    RESTARTS   AGE   IP             NODE                                                  NOMINATED NODE   READINESS GATES
sct-cluster-us-east1-b-us-east1-0   3/3     Running   1          71m   10.142.0.145   gke-long-scyl-oper-basi--default-pool-5980ee3b-j0ln   <none>           <none>
sct-cluster-us-east1-b-us-east1-1   3/3     Running   1          74m   10.142.0.149   gke-long-scyl-oper-basi--default-pool-5980ee3b-ps9v   <none>           <none>
sct-cluster-us-east1-b-us-east1-2   2/3     Running   4          24m   10.142.0.136   gke-long-scyl-oper-basi--default-pool-5980ee3b-60nw   <none>           <none>
```
"nodetool status" command output on one of the alive nodes:
```
$ kubectl exec -ti -n scylla sct-cluster-us-east1-b-us-east1-0 -c scylla -- sh
sh-4.2# nodetool status
Datacenter: us-east1-b
======================
Status=Up/Down
|/ State=Normal/Leaving/Joining/Moving
--  Address       Load       Tokens       Owns    Host ID                               Rack
UN  10.3.250.241  701.91 MB  256          ?       58e33999-bff9-444a-bdeb-5c45698f351b  us-east1
DN  10.3.242.255  312.4 MB   256          ?       b5918f73-ded5-4c68-a471-faa68e61c4a7  us-east1
UN  10.3.243.79   700.54 MB  256          ?       9a687adc-d39f-40aa-97e0-2fbe1cb7c5c8  us-east1
```

**To Reproduce**
1. Deploy "scylla" server using "scylla-operator"
2. Drain one of the dedicated "scylla" nodes
```$ kubectl drain foo-node-name -n scylla --ignore-daemonsets --delete-local-data```
3. Uncordon the "drained" node making evicted pod be replaced with the new one
```$ kubectl uncordon foo-node-name```

**Expected behavior**
New scylla-node which is part of new scylla k8s pod successfully attaches "scylla" cluster.

**Config Files**
k8s data: [k8s-data.tar.gz](https://github.com/scylladb/scylla-operator/files/6092023/k8s-data.tar.gz)

**Logs**
Please, provide `kubectl get events`, `kubectl logs -n scylla pod-name` and other relevant information.

**Environment:**
 - Platform: GKE
 - Kubernetes version: v1.17.15-gke.800
 - Scylla version: 4.2.0
 - Scylla-operator version: e.g.: v1.1.0-rc.3-0-g97a6d84

**Additional context**

**Comment by tnozicka** [comment]
Have you also restarted the backing machine? It almost looks like pvc/data-sct-cluster-us-east1-b-us-east1-2 data got lost - can you check if that's the case? For the case when node is being replace by another / looses local storage data a manual procedure setting `scylla/replace` label is needed at this point.

**Comment by vponomaryov** [comment]
@tnozicka As stated in the bug description, drain command was used with "--delete-local-data" option.
So, node stays the same, just data gets deleted. From kubernetes point of view it is not "replace", but from "scylla" point of view it is.
So, we must set "scylla/replace" label to the uncordoned node anyway?

**Comment by tnozicka** [comment]
```
      --delete-emptydir-data=false: Continue even if there are pods using emptyDir (local data that will be deleted when
the node is drained).
```
I am not sure what it actually does but we aren't using empty dir but a PVC backed by hostPath which should get deleted on ordinary reboot, nor drain.

**Comment by tnozicka** [comment]
From the dump it was setup with gce-pd default storage class. I've tried reproducing it in GCE but it worked fine for me with drain.

**Comment by vponomaryov** [comment]
Could not reproduce it using scylla-operator v1.1.0-rc.4-1-g83412a8

**Comment by vponomaryov** [comment]
We can close this issue as "cannot reproduce" assuming that something has changed and fixed it since "v1.1.0-rc.3-0-g97a6d84" till "v1.1.0-rc.4-1-g83412a8".
